extensions/sql-database-projects/src/dialogs/addSqlBindingQuickpick.ts

```diff
@@ -150,37 +152,77 @@ export async function launchAddSqlBindingQuickpick(uri: vscode.Uri | undefined,
+				// show the connection string methods (user input and connection profile options)
+				const listOfConnectionStringMethods = [constants.connectionProfile, constants.userConnectionString];
+				let connectionStringInfo;
+				while (!connectionStringInfo) {
+					const selectedConnectionStringMethod = await vscode.window.showQuickPick(listOfConnectionStringMethods, {
+						canPickMany: false,
+						title: constants.selectConnectionString,
+						ignoreFocusOut: true
+					});
+					if (!selectedConnectionStringMethod) {
+						// User cancelled
+						return;
 					}
-				) ?? '';
 
-				if (!newConnectionStringValue) {
-					// go back to select setting quickpick if user escapes from inputting the value in case they changed their mind
-					continue;
-				}
-
-				try {
-					const success = await azureFunctionsUtils.setLocalAppSetting(path.dirname(projectUri.fsPath), newConnectionStringSettingName, newConnectionStringValue);
-					if (success) {
-						connectionStringSettingName = newConnectionStringSettingName;
+					let connectionString: string = '';
+					// User chooses to enter connection string manually
+					if (selectedConnectionStringMethod === constants.userConnectionString) {
+						connectionString = await vscode.window.showInputBox(
+							{
+								title: constants.enterConnectionString,
+								ignoreFocusOut: true,
+								value: 'Server=localhost;Initial Catalog={db_name};User ID=sa;Password={your_password};Persist Security Info=False',
+								validateInput: input => input ? undefined : constants.valueMustNotBeEmpty
+							}
+						) ?? '';
+					} else {
+						// Let user choose from existing connections to create connection string from
+						let connectionUri: string = '';
+						const connectionInfo = await vscodeMssqlApi.promptForConnection(true);
+						if (!connectionInfo) {
+							// User cancelled return to selectedConnectionStringMethod prompt
+							continue;
+						}
+						try {
+							// TO DO: https://github.com/microsoft/azuredatastudio/issues/18012
+							connectionUri = await vscodeMssqlApi.connect(connectionInfo);
+							connectionStringInfo = connectionUri;
+						} catch (e) {
+							// give an error if unable to connect to selected connection and return to selectedConnectionStringMethod prompt
+							console.warn(e);
+							continue;
+						}
+						try {
+							connectionString = await vscodeMssqlApi.getConnectionString(connectionUri, false);
+						} catch (e) {
+							// go back to select setting quickpick if user escapes from inputting the value in case they changed their mind
+							console.warn(e);
+							void vscode.window.showErrorMessage(constants.failedToGetConnectionString);
+							continue;
+						}
+					}
+					if (connectionString) {
+						try {
+							const success = await azureFunctionsUtils.setLocalAppSetting(path.dirname(projectUri.fsPath), newConnectionStringSettingName, connectionString);
+							if (success) {
```

Comment:
If it's not successful we should probably inform the user before going back to the quickpick, otherwise they may get confused about why they end up back there.

Reply:
Is that not why we send out the error message in line 215 stating that it 'Failed to set connection string app setting: ' + error message of why it was unsuccessful.

Reply:
Except that you aren't throwing an exception so it's not going to hit that block